### PR TITLE
feat: allow for postgres and http functions on each extensibility point

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

After this change, users can opt to use either Postgres or HTTP functions on each extensibility/extension point. From an implementation standpoint, all  new extension points must support both HTTP and Postgres functions

### CODE REVIEW

File: internal/api/hooks.go [hunk 1]
```diff
@@ -172,67 +174,80 @@ func (a *API) runHTTPHook(ctx context.Context, r *http.Request, hookConfig conf.
...
-		if response, err = a.runHTTPHook(ctx, r, a.config.Hook.SendSMS, input, output); err != nil {
-			return internalServerError("Error invoking Send SMS hook.").WithInternalError(err)
+		if response, err = a.runHook(r, conn, a.config.Hook.SendSMS, input, output, u.Scheme); err != nil {
+			return err
 		}
-
 		if err := json.Unmarshal(response, hookOutput); err != nil {
 			return internalServerError("Error unmarshaling Send SMS output.").WithInternalError(err)
 		}
+		if hookOutput.IsError() {
```
Comment (J0) [inline internal/api/hooks.go:201]: Experimented with moving this check  into the function but am not sure how to maintain the type cast  (e.g. `hooks.SendSMSOutput`) while keeping the inputs for `runHook` generic. 

Tried using  `HookOutput` as the type for the input parameter but could not keep `runHook` generic while covering all  extension points

File: internal/api/hooks.go [hunk 2]
```diff
@@ -275,14 +291,17 @@ func (a *API) invokePostgresHook(ctx context.Context, conn *storage.Connection,
 		}
 
 		return nil
+
```
Comment (J0) [inline internal/api/hooks.go:294]:
```suggestion
```

**Review by hf — APPROVED** [review]
Looks good. Would just add Content-Type check on the HTTP response from the hook. If it's not JSON, throw an error.

File: internal/api/provider/linkedin_oidc.go
Comment (hf) [inline internal/api/provider/linkedin_oidc.go:1]: Why are these files included?

Comment (J0) [reply internal/api/provider/linkedin_oidc.go:1]: Likely accidentally committed. Thanks for the catch
